src/core/credrank/markovProcessGraph.test.js

```diff
@@ -154,6 +157,112 @@ describe("core/credrank/markovProcessGraph", () => {
         }
       });
     });
+    describe("personalAttribution", () => {
+      it("can create a virtual markov edge for each entry", () => {
+        // this tests all attribution entries, regardless of configured
+        // intervals
+        const mpg = markovProcessGraph(attributions);
+        for (const attr of attributions) {
+          for (const recipient of attr.recipients) {
+            for (const proportion of recipient.proportions) {
```

Comment:
I don't think I can approve most of these tests. Way too much logic. This is problematic for a number of reasons:
1. Who will test the tests?
2. When reading this code, the expected output is obscured from me.
3. f(x) == f(x) isn't a very strong test (duplicating the logic and then saying "it's the same whether I run it here or there" tests the environment, not the logic)

Ideally, you would define the expected edges explicitly, rather than using for loops and arithmetic. An easy hack for this is to assert the actual against an empty array, and then verify and copy the actual from the failed test output.

Reply:
I was just following the test patterns outlined in this file, I typically hardcode the expected returns but I wanted the tests to be consistent with what already exists here. Will make these changes.